### PR TITLE
Add updates and clarification to Tutorial 3

### PR DESCRIPTION
**Pull request by tir38**

### CODE REVIEW

File: samples/tutorial/Tutorial3.md [hunk 1]
```diff
@@ -221,7 +221,9 @@ object TodoEditWorkflow : StatefulWorkflow<EditProps, State, Output, TodoEditScr
 
 We want the todo edit screen to be shown when a user taps on an item on the todo list screen. To do this, we will modify the todo list workflow to show the edit screen when we are editing.
 
-The `TodoListWorkflow` will now occasionally need to render two screens instead of just the one. Its parent workflow will add the one or two screens to the backstack it constructs. Since `TodoListWorkflow`'s rendering type is `Any`, it can just return a list of screens. However, the parent workflow needs to know about the list, so it can pull the screens out and add them to its backstack. We'll change the rendering type from `Any` to `List<Any>`:
+The `TodoListWorkflow` will now occasionally need to render two screens instead of just the one. Its parent workflow `RootWorkflow` will add the one or two screens to the backstack it constructs. We need to update `TodoListWorkflow`'s rendering type to `List<Any>` so that it can return a list of screens, instead of just `TodoListScreen`. The parent workflow needs to know about the list, so it can pull the screens out and add them to its backstack.
```
Comment (tir38) [inline samples/tutorial/Tutorial3.md:224]: `TodoListWorkflow` 's render type is not `Any`. In tutorial 2 we still had it as `TodoListScreen`

File: samples/tutorial/Tutorial3.md [hunk 2]
```diff
@@ -233,63 +235,49 @@ object TodoListWorkflow : StatefulWorkflow<ListProps, State, Back, List<Any>>()
...
+        username = renderProps.username,
         todoTitles = titles,
         onTodoSelected = { context.actionSink.send(selectTodo(it)) },
         onBack = { context.actionSink.send(onBack()) }
     )
 
     return listOf(todoListScreen)
   }
+
+  private fun selectTodo(index: Int) = action {
```
Comment (tir38) [inline samples/tutorial/Tutorial3.md:249]: was missing, causing compile error

File: samples/tutorial/Tutorial3.md [hunk 3]
```diff
@@ -233,63 +235,49 @@ object TodoListWorkflow : StatefulWorkflow<ListProps, State, Back, List<Any>>()
...
+    TODO()
+  }
 }
 ```
 
-Next, update the `RootWorkflow` to pass the username into the `TodoListWorkflow` and handle the `.back` output:
```
Comment (tir38) [inline samples/tutorial/Tutorial3.md:249]: this step was done in tutorial # 2

File: samples/tutorial/Tutorial3.md [hunk 4]
```diff
@@ -233,63 +235,49 @@ object TodoListWorkflow : StatefulWorkflow<ListProps, State, Back, List<Any>>()
...
+Now that `TodoListWorkflow` renders a `List<Any>` we need to update its parent `RootWorkflow` to accept this list:
```
Comment (tir38) [inline samples/tutorial/Tutorial3.md:255]: But this did need to be done.

File: samples/tutorial/Tutorial3.md [hunk 5]
```diff
@@ -325,7 +313,45 @@ object TodoListWorkflow : StatefulWorkflow<ListProps, State, Back, List<Any>>()
 }
 ```
 
-Add actions for selecting a todo item as well as saving or discarding the changes:
+Modify `selectTodo` to transition to this new `Edit` step when the user selects a todo
```
Comment (tir38) [inline samples/tutorial/Tutorial3.md:316]: missing steps

**Review by steve-the-edwards — APPROVED** [review]
Thanks! @tir38 . LGTM

**Pull request merged**